scripts/seastar-json2code.py

```diff
@@ -316,18 +343,19 @@ def add_operation(hfile, ccfile, path, oper):
     funcs = ""
     required_query_params = []
     for param in oper.get("parameters", []):
-        if is_required_query_param(param):
-            required_query_params.append(param["name"])
-        if "enum" in param:
-            enum_decl, parse_func = generate_code_from_enum(oper["nickname"],
-                                                            param["name"],
-                                                            param["enum"])
+        query_param = Parameter(param)
+        if query_param.is_required:
+            required_query_params.append(query_param)
+        if query_param.enum is not None:
```

Comment:
why the is not none?

Reply:
@amnonh i am able to parse your question. do you mean "when it is not None"? or do you mean "why shall we check if `query_param.enum` is not None"?

Reply:
why not:
if query_param.enum:

Reply:
i see. i wanted to be very specific. as this property is `None` or "enum" is in its definition, or a list of string otherwise. an empty list should probably be handled in `generate_code_from_enum()`, and an error should be reported instead of being ignored. if i had more bandwidth, i would like to add typing annotations to this script.